src/mesh/mesh_io.jl

```diff
@@ -85,25 +85,21 @@ function save_mesh_file(mesh::TreeMesh, output_directory, timestep,
 end
 
 
-function save_mesh_file(mesh::StructuredMesh, output_directory, timestep=0)
+function save_mesh_file(mesh::CurvedMesh{NDIMS, RealT}, output_directory, timestep=0) where {NDIMS, RealT}
```

Comment:
This would break compatibility with the other `save_mesh_file` functions. In practice, this function is never called with a `timestep`, because the `CurvedMesh` should have `unsaved_changes` only before the first step, but if it were to have `unsaved_changes` in the middle of the simulation for some reason, the program would break here.

Reply:
Are you saying that with my proposed changes it would work in normal cases, but if for some reason unsaved changes is set to true, it would break? Wouldn't that be exactly what we want? That is, it works under the specified conditions, but if someone changes the program behavior in unsupported ways, it gives an error instead of silently chugging along. Feels to me that this is the safer approach, and one can always reads support for the time step of needed.

Or do you see an application where there might be unsaved changes in the current framework?

Reply:
Is there any reason to not allow users to save the mesh again?

Reply:
No, if course not. But unsaved changes does not prevent saving, it explicitly says the mesh has changed. And I don't see how this should or could happen for a structured mesh.

Reply:
Okay, we removed the parameter.